How to create an NLP application with Flair

BRIAN721 opened the pull request:
This fixes #3129 
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [ ] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [ ] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [ ] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [ ] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [ ] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [ ] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [ ] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [ ] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [ ] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [ ] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [ ] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [ ] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [ ] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

Bayler commented:
Hi @BRIAN721 Welcome to our Engineering program at Section. I will be reviewing your tutorial. 

You mentioned that you have used Google Colab, please provide a link at the end of your tutorial of your implementation. It is important to always include that in your tutorial especially if it's a code implementation. Other times, you can leverage [Replit](https://replit.com/) to share your implementation with a reader. Please provide that then I can continue with the review.

BRIAN721 commented:
Hello @Bayler, I have added the Google Colab link in the reference section. Hope that is okay

Bayler commented:
> Hello @Bayler, I have added the Google Colab link in the reference section. Hope that is okay

Awesome. I'll continue with the review.

hectorkambow commented:
plagiarism check done

Bayler commented on content/articles/how-to-create-nlp-application-with-flair/index.md:
```diff
@@ -0,0 +1,424 @@
+Flair is a natural language processing (NLP) used to build machine learning models used in text classification and speech recognition. It is used in language translation applications, speech recognition, and conversion of speech to text and text to speech applications. It's made up of a text embedding library that enables one to combine different word and document embeddings during model building ...
+
```
Flair is a natural language processing used....

This sentence is not correct. Please correct it.

Bayler commented on content/articles/how-to-create-nlp-application-with-flair/index.md:
````diff
@@ -0,0 +1,424 @@
+ 'training_utils',
+ 'visual']
+```
+
+The methods shown above such as `embeddings`, `data`, `models`, and `trainers` will be in building our model.
````
....will be in building our model?? Did you mean it will be used in building our model? Always proofread our work.

Bayler commented on content/articles/how-to-create-nlp-application-with-flair/index.md:
````diff
@@ -0,0 +1,424 @@
+import numpy as np
+```
+
+### Dataset
+The dataset used consists of a collection of offensive and non-offensive words. The dataset will be used to train our model so that it can be used to predict if a text is offensive or not.
````
The [name of the data set] used consists of.......

Please include the name of the data set in the intro sentence. Also, make sure you add the data set used in the prerequisite section.

Bayler commented on content/articles/how-to-create-nlp-application-with-flair/index.md:
```diff
@@ -0,0 +1,424 @@
+
+[CSV File of our data](https://drive.google.com/file/d/1FpEnJkw7XY3cdP1NeUzRRKGqNYsQ3Frz/view?usp=sharing)
+Download the dataset from the above link and name the file as `offensive_and_non_offensive_dataset.csv`.
+
+#### Using Pandas to read dataset
+`Pandas` will be used to import our dataset.
```
The Pandas library is used to import or read our data set? This sentence is not clear

Bayler commented on content/articles/how-to-create-nlp-application-with-flair/index.md:
````diff
@@ -0,0 +1,424 @@
+```python
+df = pd.read_csv("offensive_and_non_offensive_dataset.csv")
+```
+
+To preview our dataset and see how its structure, enables us to see all the available columns in our dataset. We use the following command.
+
````
To preview our dataset and see how its structure, enables us to see all the available columns in our data set......

What does this sentence mean? It's not clear

Bayler commented on content/articles/how-to-create-nlp-application-with-flair/index.md:
````diff
@@ -0,0 +1,424 @@
+
+```python
+df['class'].value_counts()
+```
+
+### Formatting our CSV file
````
You need to add screenshots of the outputs of these commands. Remember, this is a tutorial

Bayler commented on content/articles/how-to-create-nlp-application-with-flair/index.md:
````diff
@@ -0,0 +1,424 @@
+#### Removing one column
+We have to format our data by removing one column which we will not need. This ensures that we have the correct dataset that will be used by our model during training.
+Use this command to remove the `class` column.
+
+```python
+df1 = df[['clean_tweet','labels']]
+```
+
+The dataset will have remain with only two columns: `clean_tweet` and `labels`.
+
+#### Renaming the columns
+We have to rename our columns into simple names that we can easily understand and use.
+
+```python
+df1.columns  = ['text','labels']
+```
+
+The new column names will be `text` and `labels`.
````
Please include screenshots here too. Explaining is not enough

Bayler commented on content/articles/how-to-create-nlp-application-with-flair/index.md:
````diff
@@ -0,0 +1,424 @@
+
+This allows us to see the size of our train, test, and dev sets.
+The output is as shown.
+
+```bash
+(4671, 2)
+(2802, 2)
+(934, 2)
+(935, 2)
+```
````
Please interpret in a few words these sizes.

Bayler commented on content/articles/how-to-create-nlp-application-with-flair/index.md:
````diff
@@ -0,0 +1,424 @@
+```python
+data_folder = 'data_fst/'
+```
+
+After doing all this we can now create our corpus using the `CSVClassificationCorpus` since we are using a CSV file format hence we specify our `delimiter` as a `,`.
+
````
Consider splitting this sentence into two.

After doing all this, we can now create our corpus using the `CSVClassificationCorpus`. Since we are using a CSV file format, we specify our `delimiter` as a `,`.

Bayler commented:
Hi @BRIAN721 Thank you for submitting your first PR. I've reviewed your tutorial. A couple of revisions need to be done before I can continue for final review. Please work on them.

I've noted a couple of things from the review that I need to bring to your attention.
1. Always pass your article through the Grammarly app. Grammar errrors are unnecessary.
2. Please add punctuation marks where necessary. I've noticed the lack of commas in most of your sentences.
3. Finally, please proofread your work when finished. Some sentence are not clear.

BRIAN721 commented:
Hello @Bayler, so I have added the requested changes. Checked the grammar via Grammarly, added more explanation and screenshots. 
Kindly review
Thanks

Bayler commented:
@BRIAN721 This is an exciting article. It's in-depth and very educative. Being your first tutorial, I'm impressed. Keep up the good work and see you in the next one.

BRIAN721 commented:
Thank you for your review 
@Bayler
